File: 백준/Bronze/1934. 최소공배수/최소공배수.py
n = int(input())

for i in range(n): 
    a,b = map(int, input().split())
    n1,n2 = a,b
    
    # 유클리드 알고리즘 사용
    # a/b를 후 나머지를 a에 저장
    # a와 b의 값을 서로 바꾸고, 다시 a와 b를 나눔 - 나머지 나누기 b를 하기 위함
    # 나머지가 0될 때까지 반복 - 나머지가 0이 되면 마지막으로 나누게 된 나머지가 최대공약수
    while n2 != 0: 
        temp = n1
        n1 = n2
        n2 = temp % n2
    print(a*b//n1) # 최소 공배수    

File: 백준/Bronze/1934. 최소공배수/최소공배수.ts
const fs = require('fs');
// const [t, ...input]: string[] = ['3', '1 45000', '6 10', '13 17'];
const [t, ...input]: string[] = fs
    .readFileSync(0,'utf-8')
    .toString()
    .trim()
    .split('\n');

// 유클리드 알고리즘 사용
function getGCD(num1: number, num2: number): number {
    const currentValue = num1 % num2;
    if (currentValue === 0) {
        return num2;
    }
    return getGCD(num2, currentValue);
}

function LCM(list: string[]): void {
    if (list.length === 0) {
        return;
    }
    const numbers: number[] = (list.pop() || '') //list.pop() : 리스트의 맨마지막요소 리턴하고 삭제
        .split(' ')
        .map((v) => parseInt(v));
    console.log((numbers[0] * numbers[1]) / getGCD(numbers[0], numbers[1]));
    return LCM(list);
}

LCM(input.reverse());